### ISSUE TITLE
IBM comment 1 on 2.4.12 Accessible Name

### ISSUE THREAD

**Issue by mbgower** [w3c/wcag21#452]
IBM requests the inclusion of the word "persistent" in this SC, to read
>Where an active control has a persistent visible label…

This would allow text that may be considered to form a non-persistent label, such as the text in a title attribute, to not be included in an interpretation of concatenated accessible name. This request stems from concern that the text of title attributes is sometimes used as a fairly verbose "tip" or other form of additional information, and especially with the recommended practice of placing this information at the start of the accessible name, could have the effect of subjecting screen reader users to hearing long unnecessary content ahead of the true accessible name.

**Comment by steverep** [comment]
I think an interesting concern is raised here, but I do not think adding an undefined adjective like "persistent" is the right approach.  The onion needs to be peeled a bit further.  Here's the relevant scenarios.

#### Scenario 1: Control with a Label and Title Attribute

    <label><input type="checkbox" ... title="Please confirm you are a human">I'm not a robot</label>

In this example, the SC should be considered satisfied because the label per the WCAG definition would be considered equivalent to the text of the `<label>` element, which in turn maps to the accessible name.  The `title` attribute would nap to the accessible description instead.  See [*Accessible Name and Description Computation*](https://www.w3.org/TR/accname-aam-1.1/#mapping_additional_nd_te).  As far as this criterion is concerned, it should be treated the same as if the information was provided outside a tooltip as:

    <label><input type="checkbox" ... aria-describedby="human">I'm not a robot</label><br>
    <span id="human">Please confirm you are a human.</span>

So in this scenario, it would be desirable to clarify that the accessible description should never be considered part of the label per the WCAG definition.  Otherwise the SC would circularly force it to be part of the accessible name.  The correct way to address this might be with a Note added to the definition of "label".

#### Scenario 2: Control with Title as the Label

    <input type="text" ... title="Reason why the title should be used in 2017">

Working backwards, the `title` attribute text is the last resort in an accessible name computation.  So the question is just whether or not a tooltip alone constitutes a sufficient label per the WCAG definition.  It's actually an interesting question, but the SC is satisfied either way:
* I could argue it's not a label because the definition says it must be presented to *all* users, but anyone on a mobile browser will never see it, or
* I could argue it is a label because we have a technique like [H65](https://www.w3.org/TR/2016/NOTE-WCAG20-TECHS-20161007/H65) to satisfy other label-related criteria.

#### Concluding Question
Would a proposed note to the definition of "label" to ensure that accessible description material should never be considered a label satisfy this concern?  It would obviously also be worth documenting in Understanding.

**Comment by DavidMacDonald** [comment]
I think it would require a failure of only providing a DESCRIPTION and not a NAME.

**Comment by steverep** [comment]
> I think it would require a failure of only providing a DESCRIPTION and not a NAME.

I agree but that's not the question.  The question is: If both are provided, what does WCAG consider the "label"?  I think most would agree only the name is the label, but we need to make that clear otherwise this SC forces collapsing the description into the name.  Make sense?

**Comment by mbgower** [comment]
Thanks for your input. I believe changing the wording to 'label' helps clarify the intent, and I think this can issue can be retired.

**Comment by steverep** [comment]
Thanks @mbgower.  I do think this led to an important point though that we ought to at least put somewhere in Understanding, i.e. text that maps to the accessible description is not considered a "label" for WCAG.

**Comment by mbgower** [comment]
>  I do think this led to an important point though that we ought to at least put somewhere in Understanding, i.e. text that maps to the accessible description is not considered a "label" for WCAG.

@steverep, agreed. I'm reopening the issue to ensure that is captured. Once it has been added to the Understanding doc, it can be closed.

**Comment by awkawk** [comment]
Closing since we have the implementation follow-up label to track.
